src/reconcile/rec_col.c

```diff
@@ -963,11 +950,8 @@ __wt_rec_col_var(
             upd = upd_select.upd;
             n = WT_INSERT_RECNO(ins);
         }
+        /* Deletion of uncommitted keys are visible to all. */
```

Comment:
When walking the append list, if we don't select an update to write to disk, there must be only uncommitted updates on that key. In that case, we write a deleted key to disk and the deletion should be globally visible.